### DISCUSSION TITLE
Can classifier free guidance just use a sequence

### DISCUSSION THREAD

**Discussion by KerfuffleV2** [ggml-org/llama.cpp#3994] [Q&A] [answered]
instead of having to run a completely separate decode with its own context every step (basically halving inference speed)?

**Comment by slaren** [comment, marked as answer]
Yes, it should be possible to do this. The only reason to decode the guidance sequence on a different context would be to use a different model.

**Reply by KerfuffleV2** [reply]
Thanks! Right now, you can't even specify a different model as far as I know so that shouldn't be much of a limitation.
